src/main/java/com/github/lordrex34/config/util/TimeUtil.java

```diff
@@ -22,109 +22,231 @@
 package com.github.lordrex34.config.util;
 
 import java.time.Duration;
+import java.time.format.DateTimeParseException;
 import java.time.temporal.ChronoUnit;
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
 /**
- * @author UnAfraid
+ * @author Nik
  */
 public class TimeUtil
 {
-	public static int findIndexOfNonDigit(CharSequence text)
+	private static class Lazy
 	{
-		for (int i = 0; i < text.length(); i++)
-		{
-			if (Character.isDigit(text.charAt(i)))
-			{
-				continue;
-			}
-			return i;
-		}
-		return -1;
+		/** Pattern that matches texts similar to "1hour30min20sec" into multiple groups of digit and non-digit parts. */
+		static final Pattern PARSE_DURATION_PATTERN = Pattern.compile("(\\d+)([^\\d]+)");
 	}
-	
+
 	/**
 	 * Parses patterns like:
 	 * <ul>
+	 * <li>1millis or 10millis</li>
+	 * <li>1sec or 10secs</li>
 	 * <li>1min or 10mins</li>
 	 * <li>1day or 10days</li>
 	 * <li>1week or 4weeks</li>
-	 * <li>1month or 12months</li>
-	 * <li>1year or 5years</li>
 	 * </ul>
-	 * @param datePattern
+	 * Also multiple value types in pattern are supported as well:
+	 * <ul>
+	 * <li>1hour30min20sec</li>
+	 * <li>3days22hours20mins</li>
+	 * <li>1week5days6hours</li>
+	 * <li>10hours20mins30secs500millis200micros100nanos</li>
+	 * </ul>
+	 * Supported time units:
+	 * <ul>
+	 * <li>nanos</li>
+	 * <li>micros</li>
+	 * <li>millis</li>
+	 * <li>sec</li>
+	 * <li>secs</li>
+	 * <li>second</li>
+	 * <li>seconds</li>
+	 * <li>min</li>
+	 * <li>mins</li>
+	 * <li>minute</li>
+	 * <li>minutes</li>
+	 * <li>hour</li>
+	 * <li>hours</li>
+	 * <li>halfday</li>
+	 * <li>halfdays</li>
+	 * <li>day</li>
+	 * <li>days</li>
+	 * <li>week</li>
+	 * <li>weeks</li>
+	 * </ul>
+	 * Values such as months or years and everything above are not supported, because they are estimate values instead of precise ones.<br>
+	 * For example, one month can either be 30 or 31 days or one year can either be 365 or 366 days. A decade consists of 10 years with estimated amount of days.<br>
+	 * <br>
+	 * This method also supports the default duration pattern that is parsed by {@link Duration#parse(CharSequence)}
+	 * @param pattern the pattern of duration to be parsed.
 	 * @return {@link Duration} object converted by the date pattern specified.
 	 * @throws IllegalStateException when malformed pattern specified.
 	 */
-	public static Duration parseDuration(String datePattern)
+	public static Duration parseDuration(String pattern)
```

Comment:
This method is simply too big. That's the issue here.

I would move the switch part into either a map, an array or another (preferably static) getter method that does nothing just takes care of the switch.